apps/copilot-api/routes/auth.ts

```diff
@@ -0,0 +1,112 @@
+import express from 'express'
+import type {Request, Response} from 'express'
+import dotenv from "dotenv";
+import jwt from "jsonwebtoken";
+import {throwInternalError} from "../middleware/error.middleware";
+import {isAddress, verifyMessage} from "viem";
+import {dataSource} from "../db";
+import {AddressesEntity} from "../entities/addreesses.entity";
+import {UsersEntity} from "../entities/users.entity";
+import {v4 as uuidv4} from 'uuid'
+
+dotenv.config();
+
+const router = express.Router();
+
+
+const addressesRepo = dataSource.getRepository(AddressesEntity)
+const usersRepo = dataSource.getRepository(UsersEntity)
+
+router.post('/login', async (req: Request, res: Response) => {
+  const {signature, walletAddress, challengeMessage} = req.body;
+
+  if (!isAddress(walletAddress)) {
+    res.status(403).json({error: "Invalid wallet address"})
+    return
+  }
+
+  try {
+    const address = await verifyMessage({
+      message: challengeMessage,
+      address: walletAddress,
+      signature
+    });
+
+    if (!address) {
+      res.status(403).json({error: "Invalid signature"})
+      return
+    }
+
+
+    const existingAddress = await addressesRepo.findOne({where: {address: walletAddress}})
+
+    const user = await usersRepo.findOne({where: {uuid: existingAddress?.userId}})
+
+
+    if (!user) {
+      const newUser = await usersRepo.save({})
+      const payload = {
+        user: {
+          id: newUser.uuid
+        }
+      }
+
+      const token = jwt.sign(payload, process.env.JWT_SECRET as string, {expiresIn: '7d'})
+
+      res.status(200).send({token})
+      return;
+    }
+
+    const payload = {
+      user: {
+        id: user.uuid,
+      }
+    }
+
+    const token = jwt.sign(payload, process.env.JWT_SECRET as string, {expiresIn: '7d'})
+
+    res.status(200).send({token})
+  } catch (err) {
+
+  }
+})
+
+router.post('/wallet-address', async (req, res) => {
```

Comment:
If SIWE was to be implemented, this would not be needed, since challenge messagae would be generated on the client side with [createSIWEMessage](https://viem.sh/docs/siwe/utilities/createSiweMessage)